app/controllers/dashboard_controller.rb

```diff
@@ -659,16 +659,16 @@ def start_url_for_user(start_url)
   end
 
   def session_reset
-    # Clear session hash just to be sure nothing is left (but copy over some fields)
-    winh    = session[:winH]
-    winw    = session[:winW]
-    referer = session['referer']
+    # save some fields to recover back into session hash after session is cleared
+    keys_to_restore = [:winH, :winW, :referer, :browser, :user_TZO]
+    data_to_restore = {}
 
-    session.clear
+    keys_to_restore.each do |k|
+      data_to_restore[k] = session[k]
+    end
```

Comment:
@fryguy 

``
data_to_restore = session.slice(*keys_to_restore) 
``
throws an error
``
NoMethodError: undefined method `slice' for #<ActionController::TestSession:0x00000002d494f8>
./app/controllers/dashboard_controller.rb:665:in `session_reset'
./spec/controllers/dashboard_controller_spec.rb:314:in `block (3 levels) in <top (required)>'
``

changed it to 

``
data_to_restore = keys_to_restore.each_with_object({}) { |k, v| v[k] = session[k] }
``